src/Workspaces/Core/Portable/Workspace/Solution/RecoverableTextAndVersion.cs

```diff
@@ -16,39 +16,50 @@ namespace Microsoft.CodeAnalysis
     /// <summary>
     /// A recoverable TextAndVersion source that saves its text to temporary storage.
     /// </summary>
-    internal sealed class RecoverableTextAndVersion : ValueSource<TextAndVersion>, ITextVersionable
+    internal sealed class RecoverableTextAndVersion : ValueSource<TextAndVersion>, ITextVersionable, ITextAndVersionSource
     {
-        private readonly SolutionServices _services;
+        private readonly HostWorkspaceServices _services;
```

Comment:
We can undo this bit now, right, since #63616 went in? Because this is actually moving things away from SolutionServices and depending on Workspace types again?